### PR TITLE
[ASCII-1503] Use telemetry component inside secrets and rctelemetryreporter component

### PR DESCRIPTION
**Pull request by GustavoCaso**
### What does this PR do?

Within the Agent Share Component team we are trying to remove the usage of globals. 
The `pkg/telemetry` relays on globals to generate metrics. Although, we have a telemetry component that has the same functionality. We need to slowly migrate the usage of our telemetry metrics to use the telemetry component. 

One day we would be able to remove the `pkg/telemetry` alltogether.

In this PR started migrate some our components to that uses `pkg/telemetry` to declare a dependency on the telemetry component and use it.

Since the migration is big I decided to split the work in multiple PR.

This PR migrates:
- secrets resolver component
- rctelemetryreporter

### Additional Notes

I had to tweak the telemetry component and mock to ensure the prometheus registry is reset when the Fx app stops. Since we are going to start using the telemetry component we need to ensure the global registry is reset, mostly to ensure tests do not panic because the same metric has been registered 

### Describe how to test/QA your changes

Covered by unit tests

### CODE REVIEW

**Review by vickenty — APPROVED** [review]

**Review by mackjmr — APPROVED** [review]

File: comp/core/telemetry/telemetryimpl/telemetry.go [hunk 1]
```diff
@@ -85,8 +108,7 @@ func (t *telemetryImpl) Handler() http.Handler {
 func (t *telemetryImpl) Reset() {
 	mutex.Lock()
 	defer mutex.Unlock()
-	registry = prometheus.NewRegistry()
-	t.registry = registry
+	t.registry = newRegistry()
```
Comment (pgimalac) [inline comp/core/telemetry/telemetryimpl/telemetry.go:111]: Now the global `registry` is not reset, is that on purpose ?

File: comp/core/telemetry/telemetryimpl/telemtry_mock.go
Comment (pgimalac) [inline comp/core/telemetry/telemetryimpl/telemtry_mock.go:1]: There is a typo in the name of the file (`telemtry`)

File: comp/core/telemetry/telemetryimpl/telemetry.go (hunk 1 above)
Comment (GustavoCaso) [reply comp/core/telemetry/telemetryimpl/telemetry.go:111]: Ohh right 🤦

Comment (GustavoCaso) [reply comp/core/telemetry/telemetryimpl/telemetry.go:111]: Forgot about the global var `registry`

**Review by pgimalac — APPROVED** [review]
Nice !

**Review by ogaca-dd — APPROVED** [review]
LGTM

**Review by remeh — APPROVED** [review]

**Review by mellon85 — APPROVED** [review]

**Review by wdhif — APPROVED** [review]

**Review by ajgajg1134 — APPROVED** [review]
Looks good for APM changes

**Pull request merged**